File: Java/MethodOverrding.java
//Java Polymorphism, Method Overriding

class MethodOverrding
{
public static void main(String args[])
{
class1 class1Object = new class1();
class1Object.OverrideMethod();

class2 class2Object = new class2();
class2Object.OverrideMethod();
}
}

class class1
{
void OverrideMethod()
{
System.out.println("Method from Class1");
} 
}

class class2 extends class1
{
@Override
void OverrideMethod() /*while overriding @Override annotation is recommended but not compulsory, methods that are final cannot be overridden*/
{
System.out.println("\nMethod from Class1 Overridden by Class2");
} 
}

File: Kotlin/MethodOverrding.kt
//Kotlin Polymorphism, Method Overriding

fun main()
{
val class1Object = class1()
class1Object.OverrideMethod()

val class2Object = class2();
class2Object.OverrideMethod()
}

open class class1
{
open fun OverrideMethod()
{
println("Method from Class1")
} 
}

class class2 : class1()
{
override fun OverrideMethod() /*while overriding a method override keyword is compulsory and the method should be open*/
{
println("\nMethod from Class1 Overridden by Class2")
} 
}